rescan.go

```diff
@@ -388,13 +426,154 @@ filterHeaderWaitLoop:
 			log.Infof("Resending rescan header for block hash=%v, "+
 				"height=%v", headerTip.BlockHash(), height)
 
+			ntfn := blockntfns.NewBlockConnected(headerTip, height)
 			select {
-			case blockConnected <- headerTip:
+			case blockSubscription.Notifications <- ntfn:
 			case <-ro.quit:
 			}
 		})
 	}
 
+	// handleBlockConnected is a closure that handles a new block connected
+	// notification. The boolean returned determines whether we are synced
+	// with the tip of the chain.
+	//
+	// TODO(wilmer): refactor this and handleBlockDisconnected into their
+	// own methods.
+	handleBlockConnected := func(ntfn *blockntfns.Connected) (bool, error) {
```

Comment:
is this only a refactor, or does it have behavioral changes? It would be easier to review if extracting of logic into a helper method was done in one individual commit.